src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp

```diff
@@ -73,11 +73,45 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
         // Return elements should be one agent.
         if (response.size() == 1)
         {
-            const auto& agent = response.front();
-            data->m_agents.push_back({data->agentId().data(),
-                                      agent.at("name"),
-                                      Utils::leftTrim(agent.at("version"), "Wazuh "),
-                                      agent.at("ip")});
+            const auto clusterStatus = data->clusterStatus();
+            const auto clusterNodeName = data->clusterNodeName();
+            const auto agentId = data->agentId();
+
+            try
+            {
+                const auto& agent = response.front();
+                const auto& wdbAgentNodeName = agent.at("node_name").get_ref<const std::string&>();
+                const auto& wdbAgentConnectionStatus = agent.at("connection_status").get_ref<const std::string&>();
+                const auto& wdbAgentVersion = agent.at("version").get_ref<const std::string&>();
+                const auto& wdbAgentName = agent.at("name").get_ref<const std::string&>();
+                const auto& wdbAgentIp = agent.at("ip").get_ref<const std::string&>();
+
+                // If the agent is not part of a cluster or the agent is part of the cluster and the current agent
+                // node_name
+                // is the same as the server node name. Add the agent to the list of agents to be scanned.
+                if (!clusterStatus || (clusterNodeName == wdbAgentNodeName && "active" == wdbAgentConnectionStatus))
+                {
+                    data->m_agents.push_back(
+                        {agentId.data(), wdbAgentName, Utils::leftTrim(wdbAgentVersion, "Wazuh "), wdbAgentIp});
+                    logDebug2(
+                        WM_VULNSCAN_LOGTAG, "Agent %s added to the list of agents to be re-scanned", agentId.data());
+                }
+                else
+                {
+                    logDebug2(WM_VULNSCAN_LOGTAG,
+                              "Agent %s not added to the list of agents to be re-scanned (cluster status: %s) (cluster "
+                              "node_name: %s) (connection_status: %s) (agent node_name: %s)",
+                              agentId.data(),
+                              clusterStatus ? "true" : "false",
+                              clusterNodeName.data(),
+                              wdbAgentConnectionStatus.c_str(),
+                              wdbAgentNodeName.c_str());
+                }
+            }
+            catch (const std::exception& e)
+            {
+                throw std::runtime_error("Invalid agent-info response format");
```

Comment:
On the other catch, we return a nullptr. Shouldn't we align with that to follow the same execution path?

Reply:
make sense